js/background.js

```diff
@@ -483,84 +499,93 @@
         });
     }
 
-    function onMessageReceived(ev) {
-        var data = ev.data;
-        if (data.message.flags & textsecure.protobuf.DataMessage.Flags.PROFILE_KEY_UPDATE) {
-            var profileKey = data.message.profileKey.toArrayBuffer();
-            return ConversationController.getOrCreateAndWait(data.source, 'private').then(function(sender) {
-              return sender.setProfileKey(profileKey).then(ev.confirm);
-            });
-        }
-        var message = initIncomingMessage(data);
-
-        return isMessageDuplicate(message).then(function(isDuplicate) {
-            if (isDuplicate) {
-                console.log('Received duplicate message', message.idForLogging());
-                ev.confirm();
-                return;
-            }
-
-            var type, id;
-            if (data.message.group) {
-                type = 'group';
-                id = data.message.group.id;
-            } else {
-                type = 'private';
-                id = data.source;
-            }
-
-            return ConversationController.getOrCreateAndWait(id, type).then(function() {
-                return message.handleDataMessage(data.message, ev.confirm, {
-                    initialLoadComplete: initialLoadComplete
-                });
-            });
-        });
-    }
-
-    function onSentMessage(ev) {
-        var now = new Date().getTime();
-        var data = ev.data;
-
-        var type, id;
-        if (data.message.group) {
-            type = 'group';
-            id = data.message.group.id;
-        } else {
-            type = 'private';
-            id = data.destination;
-        }
-
-        if (data.message.flags & textsecure.protobuf.DataMessage.Flags.PROFILE_KEY_UPDATE) {
-            return ConversationController.getOrCreateAndWait(id, type).then(function(convo) {
-              return convo.save({profileSharing: true}).then(ev.confirm);
-            });
-        }
-
-        var message = new Whisper.Message({
-            source         : textsecure.storage.user.getNumber(),
-            sourceDevice   : data.device,
-            sent_at        : data.timestamp,
-            received_at    : now,
-            conversationId : data.destination,
-            type           : 'outgoing',
-            sent           : true,
-            expirationStartTimestamp: data.expirationStartTimestamp,
-        });
+  /* eslint-enable */
+  /* jshint ignore:start */
+  function createMessageHandler(
+    { createMessage, getMessageDescriptor, handleProfileUpdate }
+  ) {
+    return async (event) => {
+      const { data, confirm } = event;
+
+      const messageDescriptor = getMessageDescriptor(data);
+
+      const isProfileUpdate = Boolean(
+        // eslint-disable-next-line no-bitwise
+        data.message.flags & textsecure.protobuf.DataMessage.Flags.PROFILE_KEY_UPDATE
```

Comment:
It makes sense to split this long line up, but I think there's a better solution. We could pull that flag out into a separate variable, whether here or at the top of the file, then use it here.